walk-in-interview/src/main/webapp/update-job/script.js

```diff
@@ -0,0 +1,438 @@
+  jobPaySelectElement.value = jobPayFrequency;
+}
+
+/**
+ * Dynamically adds the options for job duration.
+ * 
+ * @param {String} jobDuration Original duration of the current job.
+ */
+function renderJobDurationOptions(jobDuration) {
+  const jobDurationSelect = document.getElementById('duration');
+  jobDurationSelect.setAttribute('required', true);
+  
+  renderSelectOptions(jobDurationSelect, STRINGS['duration']);
+  jobDurationSelect.value = jobDuration;
+}
+
+/**
+ * Dynamically adds the limits for choosing the new job post expiry.
+ * 
+ * @param {long} jobExpiryTimestamp Timestamp of the expiry date for this job post.
+ */
+function renderJobExpiryLimits(jobExpiryTimestamp) {
+  const expiryDate = new Date(jobExpiryTimestamp).toISOString().substr(0, 10);
+  
+  const date = new Date();
+  const min = date.toISOString().substr(0, 10);
+  date.setFullYear(date.getFullYear() + 1);
+  const max = date.toISOString().substr(0, 10);
+
+  const datePicker = document.getElementById('expiry');
+  datePicker.setAttribute('min', min);
+  datePicker.setAttribute('max', max);
+  datePicker.setAttribute('type', 'date');
+  datePicker.setAttribute('value', expiryDate);
+}
+
+/**
+ * Gets job detail from user input.
+ * @param {String} status Status for the current job post.
+ * @return {Object} containing the user inputs.
+ */
+function getJobDetailsFromUserInput() {
+  const name = document.getElementById('title').value;
+  const description = document.getElementById('description').value;
+  const address = document.getElementById('address').value;
+  const postalCode = document.getElementById('postal-code').value;
+  const payFrequency = document.getElementById('pay-frequency').value;
+  const payMin = document.getElementById('pay-min').valueAsNumber;
+  const payMax = document.getElementById('pay-max').valueAsNumber;
+  
+  const requirementsCheckboxes =
+    document.getElementsByName('requirements-list');
+  const requirementsList = [];
+  requirementsCheckboxes.forEach(({checked, id}) => {
+    if (checked) {
+      requirementsList.push(id);
+    }
+  });
+  
+  const expiry = document.getElementById('expiry').valueAsNumber;
+  const duration = document.getElementById('duration').value;
+
+  const jobPostId = getJobId();
+  const status = getJobFromId().jobStatus;
+
+  if (status === '') {
+    status = 'ACTIVE';
+  }
+  
+  const jobDetails = {
+    jobId: jobPostId,
+    jobStatus: status,
+    jobTitle: name,
+    jobLocation: {
+      address: address,
+      postalCode: postalCode,
+      lat: 1.3039, // TODO(issue/13): get these from places api
+      lon: 103.8358,
+    },
+    jobDescription: description,
+    jobPay: {
+      paymentFrequency: payFrequency,
+      min: payMin,
+      max: payMax,
+    },
+    requirements: requirementsList,
+    postExpiryTimestamp: expiry,
+    jobDuration: duration,
+  };
+  
+  return jobDetails;
+}
+  
+/**
+* Validates the user input
```

Comment:
Since you have side effect of setting the error message, maybe mention in the comment doc as well.